dash/include/dash/pattern/internal/PatternArguments.h

```diff
@@ -122,71 +105,91 @@ class PatternArguments {
 
 private:
   /// BlockPattern matching for extent value of type IndexType.
-  template<int count>
-  void check(SizeType extent) {
+  template<int argc_size, int argc_dist, int has_team, typename ... Args>
+  void check(SizeType extent, Args && ... args) {
+    static_assert(argc_size >= 0, "Cannot mix size and SizeSpec definition"
+        "in variadic pattern constructor!");
+    DASH_LOG_TRACE("PatternArguments.check(extent)", extent);
+    static_assert(argc_size < NumDimensions, "Number of size specifier exceeds"
+        "the number of dimensions in variadic pattern constructor!");
     DASH_LOG_TRACE("PatternArguments.check(extent)", extent);
-    _argc_size++;
-    _sizespec.resize(count, extent);
+    _sizespec.resize(argc_size, extent);
+    check<argc_size+1, argc_dist, has_team>(std::forward<Args>(args)...);
   }
   /// BlockPattern matching for up to \c NumDimensions optional 
   /// parameters specifying the distribution pattern.
-  template<int count>
-  void check(const TeamSpec_t & teamSpec) {
+  template<int argc_size, int argc_dist, int has_team, typename ... Args>
+  void check(const TeamSpec_t & teamSpec, Args && ... args) {
+    static_assert(has_team == 0,
+        "Cannot mix Team and TeamSpec definition in variadic "
+        "pattern constructor!");
```

Comment:
This limitation seems a little bit odd,
but I think this is depended on TeamSpec.